users/views.py

```diff
@@ -35,8 +35,12 @@ def get(self, request):
     def post(self, request: HttpRequest, format=None) -> HttpResponse:
         if request.method != "POST":
             return HttpResponse(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-
         request_data = JSONParser().parse(request)
+
+        #upon Prof creation, manually assert that a password was provided in the request body
+        if 'password' not in request_data['user']:
```

Comment:
If we're getting rid of the password being needed what's the point of this check?

Reply:
You need it to create a professor

Reply:
This affects only the Admin-side professor _creation_ endpoint. Logically, a Prof account must be created with a password field set, so this is an easy way to allow both cases to happen without messing around with the serializer too much. Makes sense?

Reply:
Oh silly me